app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -239,6 +244,29 @@ def update
       else
         return render status: :unprocessable_entity, json: { error: "Invalid parameter to be changed" }
       end
+    elsif group_type == UserGroup.group_types[:banned_competitors]
+      if params.key?(:endDate)
+        role.end_date = params.require(:endDate)
+      end
+
+      role.save!
+      role.previous_changes.each do |changed_key, values|
+        changed_parameter = (
+          case changed_key
+          when 'end_date'
+            'End Date'
+          else
+            nil
+          end
+        )
+        if changed_parameter.present?
+          changes << UserRole::UserRoleChange.new(
+            changed_parameter: changed_parameter,
+            previous_value: values[0] || 'None',
+            new_value: values[1] || 'None',
```

Comment:
@gregorbg Could you please review these two lines in more deep? I'm looking to print 'None' whenever the value is nil. This is working fine, but I'm confused whether it will miss any edge cases. Like, I expected `0 || 'None'` to give 'None', but it gives 0. Luckily I wanted 0. (In python it would have taken 'None') Basically I want to take 'None' only if value is `nil`.

Reply:
Yup, interpreted languages all have their own rules about what is false-y and what isn't.
Ruby is relatively strict in comparison, but also has a lot of convenience methods for you: `my_value.nil?` should do exactly what you want.

Reply:
You can de-structure the array (similar to JS `const [foo, bar] = my_awesome_array` but slightly different syntax) using `previous_value, new_value = values`. Then you can have a helper function `changed_value_to_human_readable`:

```ruby
def changed_value_to_human_readable(value)
  value.nil? ? 'None' : value
end
```

Reply:
Using the above method and destructuring, you can then do 
```suggestion
            previous_value: changed_value_to_human_readable(previous_value),
            new_value: changed_value_to_human_readable(new_value),
```
and I think that's reasonably readable

Reply:
Thanks, I've implemented your suggestion.